### PR TITLE
fix numpy docstring for function assign_step_methods

### PR DESCRIPTION
**Pull request by purna135**
Doing numpy doc validation in sampling.py

Depending on what your PR does, here are a few things you might want to address in the description:
+ [ ] what are the (breaking) changes that this PR makes?
+ [ ] important background, or details about the implementation
+ [ ] are the changes—especially new features—covered by tests and docstrings?
+ [x] [linting/style checks have been run](https://docs.pymc.io/en/latest/contributing/python_style.html)
+ [ ] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples)
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

**Comment by OriolAbril** [comment]
I can't find this function in https://docs.pymc.io/en/latest/api/samplers.html. @ricardoV94 @michaelosthege @twiecki should it be in the docs? 

@purna135 the docstring can still be updated, but we should prioritize the public api as private docstrings are read only by contributors (not by users) and in raw format only. I'll adapt the review depending on this being a public or private function.

**Comment by michaelosthege** [comment]
> I can't find this function in https://docs.pymc.io/en/latest/api/samplers.html. @ricardoV94 @michaelosthege @twiecki should it be in the docs?
> 
> @purna135 the docstring can still be updated, but we should prioritize the public api as private docstrings are read only by contributors (not by users) and in raw format only. I'll adapt the review depending on this being a public or private function.

This is internal API, but docstrings are welcome there too

**Comment by michaelosthege** [comment]
Even more important on internal API are type hints, because mypy can find bugs for us.
I'll do a PR to add some things helping us with that

**Review by michaelosthege — APPROVED** [review]

File: pymc/sampling.py [hunk 1]
```diff
@@ -181,14 +181,14 @@ def assign_step_methods(model, step=None, methods=STEP_METHODS, step_kwargs=None
     Parameters
     ----------
     model : Model object
-        A fully-specified model object
-    step : step function or vector of step functions
+        A fully-specified model object.
+    step : step function or vector of step functions, default=None
```
Comment (OriolAbril) [inline pymc/sampling.py]:
```suggestion
    step : step function or iterable of step functions, optional
```

`None` is not used directly as default value, it is a placeholder to indicate the internal defaults should be used. Therefore, `optional` should be used instead as explained in https://pymc-data-umbrella.xyz/en/latest/sprint/docstring_tutorial.html#parameters. Not sure what vector means, but it seems like iterable can be used which is much more common and recommended by numpydoc.

File: pymc/sampling.py [hunk 2]
```diff
@@ -181,14 +181,14 @@ def assign_step_methods(model, step=None, methods=STEP_METHODS, step_kwargs=None
...
         One or more step functions that have been assigned to some subset of
         the model's parameters. Defaults to ``None`` (no assigned variables).
     methods : vector of step method classes
```
Comment (OriolAbril) [inline pymc/sampling.py]:
```suggestion
    methods : iterable of step method classes, optional
```

File: pymc/sampling.py [hunk 3]
```diff
@@ -181,14 +181,14 @@ def assign_step_methods(model, step=None, methods=STEP_METHODS, step_kwargs=None
...
         The set of step methods from which the function may choose. Defaults
         to the main step methods provided by PyMC.
-    step_kwargs : dict
+    step_kwargs : dict, default=None
```
Comment (OriolAbril) [inline pymc/sampling.py]:
```suggestion
    step_kwargs : dict, optional
```

**Comment by purna135** [comment]
> I can't find this function in https://docs.pymc.io/en/latest/api/samplers.html. @ricardoV94 @michaelosthege @twiecki should it be in the docs?
> 
> @purna135 the docstring can still be updated, but we should prioritize the public api as private docstrings are read only by contributors (not by users) and in raw format only. I'll adapt the review depending on this being a public or private function.

My mistake! I assumed we needed to change all of the functions in sampling.py.
Sure, I'll work on more important tasks.

**Review by OriolAbril — APPROVED** [review]

**Comment by OriolAbril** [comment]
Thanks! Just to be extra clear, you can still work on docstrings, there are dozens left to update, but it'd be best to start choosing from https://docs.pymc.io/en/latest/api.html instead of choosing from the source code to prioritize the most impactful changes.

**Comment by purna135** [comment]
@OriolAbril, you're welcome. Sure, I'll take care of it, but I have some questions.
1. Is there any test that I can use to see if a function has a valid docstring or not, so I can choose the best method to start?
2. Is there any weekly meeting where I can participate or raise a question?

**Comment by OriolAbril** [comment]
> Is there any test that I can use to see if a function has a valid docstring or not, so I can choose the best method to start?

We don't have anything automated. You should assume all docstrings are outdated for now, choose a docstring from the [website](https://docs.pymc.io/en/latest/api.html) and then check that they are not yet listed #5459 (docstrings listed there have either been updated already or have someone working on them already)

> Is there any weekly meeting where I can participate or raise a question?

There isn't but we are trying to organize montly office hours. You should join [Discourse](https://discourse.pymc.io/) to be notified when we choose the date and the format.

**Comment by purna135** [comment]
Thank you : )

**Comment by purna135** [comment]
Hi @OriolAbril, I started working on docsting but I've noticed that several functions are missing documentation.
like: [Flat.get_moment](https://docs.pymc.io/en/latest/api/distributions/generated/classmethods/pymc.Flat.get_moment.html#pymc.Flat.get_moment)

Is it necessary to include a description of these functions?

**Comment by OriolAbril** [comment]
No, you should not add any new docstrings anywhere, only make sure the ones we have follow numpydoc

**Pull request merged**